src/main/java/com/linecorp/armeria/client/HttpRemoteInvoker.java

```diff
@@ -143,8 +117,27 @@
         return resultPromise;
     }
 
-    private EventLoop eventLoop() {
-        return ServiceInvocationContext.mapCurrent(ServiceInvocationContext::eventLoop, eventLoopGroup::next);
+    private KeyedChannelPool<PoolKey> pool(EventLoop eventLoop) {
+        KeyedChannelPool<PoolKey> pool = map.get(eventLoop);
+        if (pool != null) {
```

Comment:
Probably to avoid the instantiation of the `Function` that's given as the second argument?